### PR TITLE
Hotfix to add VEM as soft dependency to `erroranalysis`

### PR DESCRIPTION
**Pull request by Advitya17**
Hotfix to add VEM as soft dependency to `erroranalysis` to not require users to have torch for tabular scenarios.

## Description

VEM (and hence torch) was becoming a hard dependency for `erroranalysis` and hence tabular RAI dashboard:

> from responsibleai import RAIInsights
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/responsibleai/__init__.py:8: in <module>
    from responsibleai.modelanalysis import ModelAnalysis
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/responsibleai/modelanalysis/__init__.py:8: in <module>
    from responsibleai.modelanalysis.model_analysis import ModelAnalysis
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/responsibleai/modelanalysis/model_analysis.py:15: in <module>
    from responsibleai.managers.error_analysis_manager import ErrorAnalysisManager
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/responsibleai/managers/error_analysis_manager.py:15: in <module>
    from erroranalysis._internal.error_analyzer import ModelAnalyzer
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/erroranalysis/_internal/error_analyzer/__init__.py:5: in <module>
    from erroranalysis.analyzer import ModelAnalyzer, PredictionsAnalyzer
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/erroranalysis/analyzer/__init__.py:5: in <module>
    from .error_analyzer import ModelAnalyzer, PredictionsAnalyzer
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/erroranalysis/analyzer/error_analyzer.py:12: in <module>
    from vision_explanation_methods.error_labeling.error_labeling import \
/usr/share/miniconda/envs/test/lib/python3.9/site-packages/vision_explanation_methods/error_labeling/error_labeling.py:10: in <module>
    import torchvision
E   ModuleNotFoundError: No module named 'torchvision'

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: erroranalysis/erroranalysis/_internal/matrix_filter.py [hunk 1]
```diff
@@ -8,8 +8,14 @@
...
 import pandas as pd
 from sklearn.metrics import multilabel_confusion_matrix
-from vision_explanation_methods.error_labeling.error_labeling import \
-    ErrorLabeling
+
+try:
+    from vision_explanation_methods.error_labeling.error_labeling import \
+        ErrorLabeling
+except ImportError:
+    warnings.warn("Can't import vision_explanation_methods"
```
[the remark this answers is not in the document]
Comment (Advitya17) [reply erroranalysis/erroranalysis/_internal/matrix_filter.py:16]: For that we'd need access to the `model_task` / `analyzer`, which doesn't seem to be available globally :/

Any other suggestions on how we could do this ... ?

[the remark this answers is not in the document]
Comment (imatiach-msft) [reply erroranalysis/erroranalysis/_internal/matrix_filter.py:16]: We should do this like with other similar imports - set a flag on whether torch_installed = True after import, and in failure case set torch_installed = False.
Then, inside the code below, check the condition and throw if torch_installed == False.

**Review by gaugup — APPROVED** [review]
How are you planning to test the vision error analysis flow?

**Comment by Advitya17** [comment]
> How are you planning to test the vision error analysis flow?

Some test cases were already added in https://github.com/microsoft/responsible-ai-toolbox/pull/2225/files#diff-db619f3fdbc5b12b9ebb0d194ced62df8b1de699428f71e9be372f73f419d7e6. Plan to try adding more.

File: erroranalysis/setup.py [hunk 2]
```diff
@@ -15,6 +15,10 @@
 with open('requirements.txt') as f:
     install_requires = [line.strip() for line in f]
 
+extras_require = {
```
Comment (imatiach-msft) [inline erroranalysis/setup.py:18]: do you need to add this to the error analysis CI (github action) to run the tests? I'm guessing it's skipping the test now?

**Review by imatiach-msft — APPROVED** [review]

File: erroranalysis/setup.py (hunk 2 above)
Comment (Advitya17) [reply erroranalysis/setup.py]: Great catch, thank you!

File: erroranalysis/tests/test_metrics.py [hunk 3]
```diff
@@ -54,6 +67,8 @@ def test_multiclass_classification_metrics(self, metric):
             metric_value = metric_to_func[metric](y_true, y_pred)
         assert isinstance(metric_value, float)
 
+    @pytest.mark.skipif(not vem_installed,
```
[the remark this answers is not in the document]
Comment (Advitya17) [reply erroranalysis/tests/test_metrics.py:70]: @romanlutz I had the same question, is there a way to test the package with and without installing extras? Do I need a new PR gate, or perhaps it's done for some other package and I can emulate that .. ?

[the remark this answers is not in the document]
Comment (romanlutz) [reply erroranalysis/tests/test_metrics.py:70]: It's coupled with the CI workflow for other packages, so you might need to make one for just error analysis. I would set up a matrix strategy for this, like https://github.com/microsoft/responsible-ai-toolbox/blob/e2665874360291a5c08414a23b06beace67a64be/.github/workflows/CI-python.yml#L17. Basically add another dimension of the matrix for what to install `["erroranalysis", "erroranalysis[object_detection]"]` and your tests should be smart enough to skip if they're not needed in one case or the other. That way you can add both kinds of tests.

**Pull request merged**